source/develop/release-management/features/network/ipv6_subnets.html.md

```diff
@@ -152,21 +374,153 @@ provision the following networking API entities, represented in yaml:
     cidr: "bef0:1234:a890:5678::/64"
     enable_dhcp: yes
     gateway_ip: bef0:1234:a890:5678::1
+    ipv6_address_mode: dhcpv6_stateful
   register: ipv6_subnet
 
-- name: create a logical switch port
-  os_port:
+- name: create a router to send RA messages
+  os_router:
     cloud: "{{ cloud_name }}"
     state: present
-    name: lport1
-    network: "{{ ipv6_network.id }}"
-    fixed_ips:
-      - subnet_id: "{{ ipv6_subnet.id }}"
+    name: router0
+    interfaces:
+      - "{{ ipv6_subnet.id }}"
 ~~~~~
 
+### Configuring stateless DHCPv6 on the subnet
+
+To configure stateless dhcpv6 on the subnet, the user should provision
+the following yaml based ansible playbook snippet:
+
+~~~~~
+- name: create a network
+  os_network:
+    cloud: "{{ cloud_name }}"
+    state: present
+    name: netv6
+  register: ipv6_network
+
+- name: attach a subnet to the logical network
+  os_subnet:
+    cloud: "{{ cloud_name }}"
+    state: present
+    network_name: netv6
+    name: subnet_ipv6
+    ip_version: 6
+    cidr: bef0:1234:a890:5678::/64
+    gateway_ip: bef0:1234:a890:5678::1
+    ipv6_address_mode: dhcpv6_stateless
+  register: dhcpv6_subnet
+
+- name: create a router to send RA messages
+  os_router:
+    cloud: "{{ cloud_name }}"
+    state: present
+    name: router0
+    interfaces:
+      - "{{ dhcpv6_subnet.id }}"
+~~~~~
+
+## UI considerations
+
+The subnet ipv6_address_mode parameter should be added to the UI, since
+updates to existing subnets are not possible through the UI, and updates to the
+ipv6_address_mode parameter are not allowed, as per the
+[networking API](https://developer.openstack.org/api-ref/network/v2/?expanded=create-subnet-detail,update-subnet-detail#update-subnet). Until the UI is updated,
+the subnet should be created via ovirt-provider-ovn RESTful API or ansible.
+
+It is outside the scope of this feature to implement the UI changes that allow
+logical routers to be provisioned, and the REST API - or its ansible wrapper
+should be used instead. An ansible example is available
+[here](#configuring-stateless-dhcpv6-on-the-subnet).
+
 ## Testing
 
-Two scenarios are recommended:
-  - single IPv6 subnet
-  - multiple IPv6 subnets connected by router
+Two scenarios are recommended, each with two different configurations, leading
+to four different tests:
+  - single IPv6 subnet using dhcpv6-stateful configuration
+  - single IPv6 subnet using dhcpv6-stateless configuration
+  - multiple IPv6 subnets connected by router using dhcpv6-stateful configuration
+  - multiple IPv6 subnets connected by router using dhcpv6-stateless configuration
+
+All four scenarios on EL8 guest VMs require a router, whereas on EL7 guest VMs
+only the dhcpv6-stateless configurations require it.
```

Comment:
At least the first two tests are required for other guests than RHEL8, too. I think at least the latest supported Windows Server would important to test.

Reply:
The whole RFE is about adding support to EL8 guests, which, IMHO makes this out of scope.

Reply:
@mburman5 What is your view on this?
Should other clients than your default config (dhclient) tested, too?
From my point of view, Windows Server, and network manager internal, and maybe dhcpcd via network manger are worth to be checked if they are working.

Reply:
My view on this is very simple, i(we) only care for the rhel scenarios(rhel7.z + rhel8).
The default dhclient is good enough for us. We will test dhclient and NM scenarios i guess.

Reply:
ok, please note that the network manager in your image is explicitly configured to use dhclient, which is not the default configuration of the public RHEL8 beta.

Reply:
Thanks for raising that concern Dominik. My 2  cents is that we should - at least - test the default vanilla config.

Can't we (also) use a vanilla rhel8 template ?

Reply:
> Can't we (also) use a vanilla rhel8 template ?

If you asked me, from my point of view, this sounds great, we just have to learn how to get this.

Reply:
@dominikholler so what is the default configuration of the public beta? 
Isn't dhclient must be the default?

Reply:
> what is the default configuration of the public beta?
> Isn't dhclient must be the default?

@cathay4t can you help us here?

Reply:
I found the answer myself:
dhclient is installed, but not used in default config by NetworkManager in RHEL8

Reply:
https://github.com/NetworkManager/dhcp/blob/d2ca21d6f8acca0f2228f82167cda3b0fca631df/includes/site.h#L282 :
#define DHCLIENT_DEFAULT_PREFIX_LEN 64

Reply:
Yup, that's currently it. 

Never the less, if you check upstream dhclient code, it already is using a /128 prefix, and even the comment above the line you've shown above says:
"he two most useful values are 128 which is **what the current specifications call for** or 64 which is **what has been used in the past**".

Thus, to me it seems the vanilla configuration will work as seamlessly as the centOS 7 version currently is, but, the implementation of stateless dhcp is also necessary - to adapt to the current specifications, and to enable the MTU configuration.

Reply:
Sure, no doubts about it.

Reply:
resolving the conversation then.